Decorator_pattern/app/src/main/java/com/example/decorator_pattern/Coffee.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.decorator_pattern
+
+
+class ChocoCoffee(private var coffee:Beverage) : Decorator() {
```

Comment:
```suggestion
class Choco or ChocoDecorator or ChocoWith??(private var coffee:Beverage) : Decorator() {
```
component는 decorator의 생성인자에 따라 달라집니다.
따라서, component인 coffee 를 클래스 이름으로 활용하는것보단 decorator 관점에서 적절한 이름을 사용하는것이 좋지않을까요?

```kotlin
val americano = BaseCoffee()
val latte = MilkWith(americano)
val mochaLatte = ChocoWith(latte)
```